src/Font/UiAttributedString.cc

```diff
@@ -0,0 +1,107 @@
+#include "../../ui.h"
+#include "../ui-node.h"
+#include "nbind/nbind.h"
+
+UiAttributedString::UiAttributedString(uiAttributedString *str) {
+	s = str;
+}
+
+UiAttributedString::UiAttributedString(const char *str) {
+	s = uiNewAttributedString(str);
+}
+
+void UiAttributedString::free() {
+	uiFreeAttributedString(s);
+}
+
+uiAttributedString *UiAttributedString::getHandle() {
+	return s;
+}
+
+const char * UiAttributedString::toString() {
+	return uiAttributedStringString(s);
+}
+
+size_t UiAttributedString::toStringLen() {
+	return uiAttributedStringLen(s);
+}
+
+void UiAttributedString::appendUnattributed(const char *str) {
+	uiAttributedStringAppendUnattributed(s, str);
+}
+
+void UiAttributedString::insertUnattributed(const char *str, size_t at) {
+	uiAttributedStringInsertAtUnattributed(s, str, at);
+}
+
+void UiAttributedString::deleteString(size_t start, size_t end) {
+	uiAttributedStringDelete(s, start, end);
+}
+
+void UiAttributedString::setAttribute(UiFontAttribute *attr, size_t start, size_t end) {
+	uiAttributedStringSetAttribute(s, attr->getHandle(), start, end);
+}
+
+static unsigned int UiAttributedString__forEach(const uiAttributedString *s, const uiAttribute *a, size_t start, size_t end, void *data) {
+	nbind::cbFunction *cb = (nbind::cbFunction *) data;
+
+	return cb->call<unsigned int>(
+		UiAttributedString((uiAttributedString*)s),
+		UiFontAttribute((uiAttribute*)a),
+		start, end);
+}
+
+void UiAttributedString::forEach(nbind::cbFunction& cb) {
+
+	uiAttributedStringForEachAttribute(s, UiAttributedString__forEach, &cb);
+}
+
+
+
+void UiAttributedString::appendAttributed(const char *str, UiFontAttribute *attr) {
+	this->appendAttributed(str, attr, nullptr);
+}
+
+
+void UiAttributedString::appendAttributed(const char *str, UiFontAttribute *attr, UiFontAttribute *attr2) {
+	size_t start = this->toStringLen();
+	// TODO how this (and strlen) work with unicode?
+	size_t end = start + strlen(str);
+
+	this->appendUnattributed(str);
+	this->setAttribute(attr, start, end);
+	if(attr2 != nullptr){
+		this->setAttribute(attr2, start, end);
+	}
+}
+
+
+size_t UiAttributedString::numGraphemes() {
+	return uiAttributedStringNumGraphemes(s);
+}
+
+size_t UiAttributedString::byteIndexToGrapheme(size_t pos) {
+	return uiAttributedStringByteIndexToGrapheme(s, pos);
+}
+
+size_t UiAttributedString::graphemeToByteIndex(size_t pos) {
+	return uiAttributedStringGraphemeToByteIndex(s, pos);
+}
+
+NBIND_CLASS(UiAttributedString) {
+	construct<const char *>();
+	method(free);
+	method(toString);
+	method(toStringLen);
+	method(appendUnattributed);
+	method(insertUnattributed);
+	method(forEach);
+	multimethod(appendAttributed, args(const char *, UiFontAttribute *));
+	multimethod(appendAttributed, args(const char *, UiFontAttribute *, UiFontAttribute *), "appendAttributed2");
```

Comment:
Renaming the function for the JS side works, but for Javascript, they still aren't really overloaded.
I would also like to overload
```cpp
UiFontAttribute.newUnderlineColor(libui.textUnderlineColor.grammar, new libui.Color(0, 0, 0, 0))
UiFontAttribute.newUnderlineColor(libui.textUnderlineColor.custom, new libui.Color(0.3, 0.3, 0.5, 0.3))
```
because the color is only used with `custom`.

Reply:
It doesn't work for me neither... I opened an issue on nbind repo:
https://github.com/charto/nbind/issues/113